pkg/kubelet/kubelet_node_status.go

```diff
@@ -227,6 +227,18 @@ func (kl *Kubelet) initialNode() (*v1.Node, error) {
 		}
 		nodeTaints = append(nodeTaints, taints...)
 	}
+
+	// If TaintNodesByCondition enabled, taint node with TaintNodeUnschedulable when initializing
+	// node to avoid race condition; refer to #63897 for more detail.
+	if utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition) {
+		if node.Spec.Unschedulable {
+			nodeTaints = append(nodeTaints, v1.Taint{
```

Comment:
Should this verify that nodeTaints doesn't already include Unschedulable before adding it?

Reply:
yes. updated :)